src/main/java/org/kohsuke/github/GHRateLimit.java

```diff
@@ -221,44 +264,112 @@ public int hashCode() {
+     * @return a merged {@link GHRateLimit} with the latest {@link Record}s from these two instances. If the merged
+     *         instance is equal to the current instance, the current instance is returned. If the merged instance is
+     *         equal to the current instance, the current instance is returned.
+     */
+    @Nonnull
+    GHRateLimit getMergedRateLimit(@Nonnull GHRateLimit newLimit) {
+
+        GHRateLimit merged = new GHRateLimit(getCore().currentOrUpdated(newLimit.getCore()),
+                getSearch().currentOrUpdated(newLimit.getSearch()),
+                getGraphQL().currentOrUpdated(newLimit.getGraphQL()),
+                getIntegrationManifest().currentOrUpdated(newLimit.getIntegrationManifest()));
+
+        if (merged.equals(this)) {
+            merged = this;
+        }
+
+        return merged;
+    }
+
+    /**
+     * Gets the specified {@link Record}.
+     *
+     * {@link RateLimitTarget#NONE} will return {@link UnknownLimitRecord#DEFAULT} to prevent any clients from
+     * accidentally waiting on that record to reset before continuing.
      *
-     * @param urlPath
-     *            the url path of the request
-     * @return the {@link Record} for a url path.
+     * @param rateLimitTarget
+     *            the target rate limit record
+     * @return the target {@link Record} from this instance.
      */
     @Nonnull
-    Record getRecordForUrlPath(@Nonnull String urlPath) {
-        if (urlPath.equals("/rate_limit")) {
-            return new UnknownLimitRecord();
-        } else if (urlPath.startsWith("/search")) {
+    Record getRecord(@Nonnull RateLimitTarget rateLimitTarget) {
+        if (rateLimitTarget == RateLimitTarget.CORE) {
+            return getCore();
+        } else if (rateLimitTarget == RateLimitTarget.SEARCH) {
             return getSearch();
-        } else if (urlPath.startsWith("/graphql")) {
+        } else if (rateLimitTarget == RateLimitTarget.GRAPHQL) {
             return getGraphQL();
-        } else if (urlPath.startsWith("/app-manifests")) {
+        } else if (rateLimitTarget == RateLimitTarget.INTEGRATION_MANIFEST) {
             return getIntegrationManifest();
+        } else if (rateLimitTarget == RateLimitTarget.NONE) {
+            return UnknownLimitRecord.DEFAULT;
         } else {
-            return getCore();
+            throw new IllegalArgumentException("Unknown rate limit target: " + rateLimitTarget.toString());
         }
     }
 
     /**
      * A limit record used as a placeholder when the the actual limit is not known.
-     * <p>
-     * Has a large limit and long duration so that it will doesn't expire too often.
      *
      * @since 1.100
      */
     public static class UnknownLimitRecord extends Record {
 
-        // One hour
-        private static final long unknownLimitResetSeconds = 60L * 60L;
+        private static final long defaultUnknownLimitResetSeconds = Duration.ofSeconds(30).toMillis() / 1000;
```

Comment:
this seems a bit strange? 

```suggestion
        private static final long defaultUnknownLimitResetSeconds = Duration.ofSeconds(30).getSeconds();
```

Reply:
@timja 
There is a `Duration.toSeconds();` but only in Java 11+.

`getSeconds()` is not quite the same as `toSeconds()`, but you are correct for this case it can be considered the same.